src/main/cpp/FfmpegJavaAvPlaybackPipline.cpp

```diff
@@ -0,0 +1,368 @@
+	if (pJavaPlayback == nullptr) {
+		return ERROR_PLAYER_NULL;
+	}
+	// TODO(fraudies): Implement this
+	*width = pJavaPlayback->get_image_width();
+
+	return ERROR_NONE;
+}
+
+uint32_t FfmpegJavaAvPlaybackPipline::GetImageHeight(int* iHeight) const {
+	if (pJavaPlayback == nullptr) {
+		return ERROR_PLAYER_NULL;
+	}
+	// TODO(fraudies): Implement this
+	*iHeight = pJavaPlayback->get_image_height();
+
+	return ERROR_NONE;
+}
+
+uint32_t FfmpegJavaAvPlaybackPipline::GetAudioFormat(AudioFormat* pAudioFormat) const {
+	if (pJavaPlayback == nullptr) {
+		return ERROR_PLAYER_NULL;
+	}
+	// TODO(fraudies): Implement this
+	*pAudioFormat = pJavaPlayback->get_audio_format();
+
+	return ERROR_NONE;
+}
+
+uint32_t FfmpegJavaAvPlaybackPipline::GetPixelFormat(PixelFormat* pPixelFormat) const {
+	if (pJavaPlayback == nullptr) {
+		return ERROR_PLAYER_NULL;
+	}
+	// TODO(fraudies): Implement the pixel format for the ouput
+	*pPixelFormat = pJavaPlayback->get_pixel_format();
+
+	return ERROR_NONE;
+}
+
+uint32_t FfmpegJavaAvPlaybackPipline::GetImageBuffer(uint8_t** ppImageBuffer) {
+	if (pJavaPlayback == nullptr) {
+		return ERROR_PLAYER_NULL;
+	}
+	// TODO(fraudies): Implement the image buffer data
+	return ERROR_NONE;
+}
+
+uint32_t FfmpegJavaAvPlaybackPipline::GetAudioBuffer(uint8_t** ppAudioBuffer) {
+	if (pJavaPlayback == nullptr) {
+		return ERROR_PLAYER_NULL;
+	}
+	// TODO(fraudies): Implement the audio buffer data
+	return ERROR_NONE;
+}
+
+void FfmpegJavaAvPlaybackPipline::UpdatePlayerState(PlayerState newState) {
+	PlayerState newPlayerState = m_PlayerState;	// If we assign the same state again
+	bool bSilent = false;
+
+	switch (m_PlayerState)
+	{
+	case Unknown:
+		if (Ready == newState)
+		{
+			newPlayerState = Ready;
+		}
+		break;
+
+	case Ready:
+		if (Playing == newState)
+		{
+			newPlayerState = Playing;
+		}
+		break;
+
+	case Playing:
+		if (Stalled == newState || Paused == newState || Stopped == newState || Finished == newState) {
+			newPlayerState = newState;
+		}
+		break;
+
+	case Paused:
+		if (Stopped == newState || Playing == newState)
+		{
+			newPlayerState = newState;
+		}
+		break;
+
+	case Stopped:
+		if (Playing == newState || Paused == newState)
+		{
+			newPlayerState = newState;
+		}
+		break;
+
+	case Stalled:
+	{
+		if (Stopped == newState || Paused == newState || Playing == newState) {
+			newPlayerState = newState;
+		}
+		break;
+	}
+
+	case Finished:
+		if (Playing == newState) {
+			// We can go from Finished to Playing only when seek happens (or repeat)
+			// This state change should be silent.
+			newPlayerState = Playing;
+			bSilent = true;
+		}
+		if (Stopped == newState) {
+			newPlayerState = Stopped;
+		}
+
+		break;
+
+	case Error:
+		break;
+	}
+
+	// The same thread can acquire the same lock several times
+	SetPlayerState(newPlayerState, bSilent);
+}
+
+
+void FfmpegJavaAvPlaybackPipline::SetPlayerState(PlayerState newPlayerState, bool bSilent) {
```

Comment:
Add TODO: Push this into a super-class that is common to the SDL/Java implementation

Reply:
Added.